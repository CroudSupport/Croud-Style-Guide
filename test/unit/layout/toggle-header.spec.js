
import Vue from 'vue'
import toggleHeader from '../../../src/components/shared/layout/toggle-header'

const Constructor = Vue.extend(toggleHeader)

const vm = new Constructor({
    propsData: {
        title: 'Test Header',
        collapsed: false,
    },
}).$mount()

describe('toggle-header', () => {
    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
    })

    it('should show the collapse button', () => {
        expect(vm.$el.querySelectorAll('.large.angle.down.icon').length).toBe(1)
    })
})

describe('collapsed is false', () => {
    beforeEach(() => {
        vm.collapsed = true
    })

    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
    })

    it('should show the expand button', () => {
        expect(vm.$el.querySelectorAll('.large.angle.right.icon').length).toBe(1)
    })
})
