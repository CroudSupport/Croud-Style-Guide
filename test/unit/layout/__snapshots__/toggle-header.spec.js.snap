// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collapsed is false should match the snapshot 1`] = `
<div
  class="collapsable-header"
>
  <div
    class="title"
  >
    Test Header
  </div>
  <div
    class="caret"
  >
    <i
      class="large angle right icon"
    />
  </div>
</div>
`;

exports[`toggle-header should match the snapshot 1`] = `
<div
  class="collapsable-header"
>
  <div
    class="title"
  >
    Test Header
  </div>
  <div
    class="caret"
  >
    <i
      class="large angle down icon"
    />
  </div>
</div>
`;
